src/SFA.DAS.Admin.Aan.Web/Controllers/CreateEvent/GuestSpeakerController.cs

```diff
@@ -0,0 +1,94 @@
+﻿using FluentValidation;
+using FluentValidation.AspNetCore;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.Admin.Aan.Application.Services;
+using SFA.DAS.Admin.Aan.Web.Infrastructure;
+using SFA.DAS.Admin.Aan.Web.Models.NetworkEvent;
+
+namespace SFA.DAS.Admin.Aan.Web.Controllers.CreateEvent;
+[Authorize]
+[Route("manage-events/new/guest-speaker")]
+public class GuestSpeakerController : Controller
+{
+    private readonly ISessionService _sessionService;
+    private readonly IValidator<GuestSpeakerAddViewModel> _validator;
+
+    public const string ViewPath = "~/Views/NetworkEvent/GuestSpeakerAdd.cshtml";
+    public GuestSpeakerController(ISessionService sessionService, IValidator<GuestSpeakerAddViewModel> validator)
+    {
+        _sessionService = sessionService;
+        _validator = validator;
+    }
+
+    [HttpGet]
+    [Route("add", Name = RouteNames.CreateEvent.GuestSpeakerAdd)]
+    public IActionResult Get(GuestSpeakerAddViewModel model)
+    {
+        var augmentedModel = GetViewModel(model);
+
+        return View(ViewPath, augmentedModel);
+    }
+
+
+    [HttpGet]
+    [Route("delete", Name = RouteNames.CreateEvent.GuestSpeakerDelete)]
+    public IActionResult Delete(int id)
+    {
+        var sessionModel = _sessionService.Get<CreateEventSessionModel?>();
+        if (sessionModel == null) return RedirectToAction("Get", "NetworkEventFormat");
```

Comment:
This should be done in a filter so that the same check happens in GET as well as POST. The filter is already implemented in Apprentice and Employer web